vm.go

```diff
@@ -139,9 +153,21 @@ func CreateVirtualMachineSpec(opts *VirtualMachineOptions) (*VirtualMachineSpec,
 	}
 
 	if opts.UseGuestConnection {
-		spec.VirtualMachine.GuestConnection = &hcsschema.GuestConnection{
-			UseVsock:            opts.GuestConnectionUseVsock,
-			UseConnectedSuspend: true,
+		if !opts.ExternalGuestConnection {
+			// gcs connects to hcs (internal)
+			spec.VirtualMachine.GuestConnection = &hcsschema.GuestConnection{
+				UseVsock:            opts.GuestConnectionUseVsock,
+				UseConnectedSuspend: true,
+			}
+		} else {
+			// gcs connects to hcsshim (external)
+			// Allow administrators and SYSTEM to bind to vsock sockets
+			// so that we can create GCS sockets.
+			spec.VirtualMachine.Devices.HvSocket = &hcsschema.HvSocket2{
+				HvSocketConfig: &hcsschema.HvSocketSystemConfig{
+					DefaultBindSecurityDescriptor: "D:P(A;;FA;;;SY)(A;;FA;;;BA)",
```

Comment:
where do we get this value? is this documented?

Reply:
It's acl in string format, meaning of content is expressed in comment "to Allow administrators and SYSTEM to bind to vsock sockets"